File: mo_lucro_app/lib/models/portfolio_position.dart
/// Represents a consolidated portfolio position for a single asset.
class PortfolioPosition {
  final String asset;
  final String category;
  final double quantity;
  final double totalInvested;
  final double avgPrice;

  const PortfolioPosition({
    required this.asset,
    required this.category,
    required this.quantity,
    required this.totalInvested,
    required this.avgPrice,
  });

  /// Since we don't have live prices, using avgPrice as current (0% P&L baseline)
  /// Replace [currentPrice] with a live API value when available.
  double profitLoss(double currentPrice) =>
      (currentPrice - avgPrice) * quantity;

  double profitLossPercent(double currentPrice) =>
      avgPrice > 0 ? ((currentPrice - avgPrice) / avgPrice) * 100 : 0;

  double currentValue(double currentPrice) => currentPrice * quantity;

  @override
  String toString() =>
      'PortfolioPosition(asset: $asset, qty: $quantity, avg: $avgPrice, invested: $totalInvested)';
}

File: mobile/src/models/portfolioPosition.ts
/** Porte de lib/models/portfolio_position.dart — posição consolidada de um ativo. */

import type { AssetCategory } from './operation';

export type PortfolioPosition = {
  asset: string;
  category: AssetCategory;
  quantity: number;
  totalInvested: number;
  avgPrice: number;
};

export function profitLoss(position: PortfolioPosition, currentPrice: number): number {
  return (currentPrice - position.avgPrice) * position.quantity;
}

export function profitLossPercent(position: PortfolioPosition, currentPrice: number): number {
  if (position.avgPrice <= 0) return 0;
  return ((currentPrice - position.avgPrice) / position.avgPrice) * 100;
}

export function currentValue(position: PortfolioPosition, currentPrice: number): number {
  return currentPrice * position.quantity;
}
